src/openexr.imageio/exrinput.cpp

```diff
@@ -319,7 +320,9 @@ void set_exr_threads ()
 
     int oiio_threads = 1;
     OIIO::getattribute ("exr_threads", oiio_threads);
-
+    if (oiio_threads == 0) {
+        oiio_threads = Sysutil::hardware_concurrency();
```

Comment:
I think we should use hardware_concurrency everywhere indeed, as said this is the responsibility of the client to use it correctly.
I'll benchmark 12vs24 next week if I get time and get back on the topic, but I'm pretty sure OpenEXR handles this well because there's only ever 1 thread doing system I/O in the library